### PR TITLE
Slime plushes, rainbow flush plush and related changes

### PR DESCRIPTION
**Pull request by Gilgaxx**
## About The Pull Request
This PR adds a total of 6 new plushes (slimeplush_pink, slimeplush_green, slimeplush_blue, slimeplush_red, slimeplush_rainbow and flushplush_rainbow)
fCaWh3rS82
Added said plushes to relevant places:
* Plush order beacon
* Arcade and Warping gold rune (added the flushplush_rainbow and the new slimeplushie/random to both of these)
* Cargo plush crate (added new plushs to the normal plush crate, removed the dehydrated space carp form the normal crate, added a new contraband only crate without moths but with the dehydrated carp (Plushie Crate Without Moth Plushies))
* Mail (RD and scientists can get slime plushes, curators can get the flushplush_rainbow rarely)
* Family heirloom (added the new slime variants for RD and scientist)
* Phobia (Alien phobia has the slime mob as part of the phobia but not the slime plush, so that was added (to match the lizard phobia working with the lizard plushie))

Also made some minor grammar fixes to the moth plush descriptions.

## Why It's Good For The Game

Before this PR, getting a random plush usually meant getting a large number of moth plushes (25 out of 43 plushes). So by adding more slime plushes and a new flush plush, we lower the odds of getting the moth plushes. 
The addition of the crate is seen as a fix to the original containing contraband, adding a new contraband crate without the moth plushes.
Along the way, put the new slime plushes in someplaces, put slime plushes in places that they weren't before (like the arcade and phobia), also fixing grammar is always nice :)

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

* Warping gold rune
dreamseeker_YFkmxxkSNZ

* Cargo plush crate - Before emagging the console
dreamseeker_8R0lqgLLPn
After emagging the console
dreamseeker_ADBHZyIAmC
Opened crate
dreamseeker_wUq9qztSA1

* Mail - RD opening the mail (and getting a green slime)
dreamseeker_8GucNSKleE

* Family heirloom - Spawning as a scientist (also getting a green slime)
dreamseeker_pjNJSqHGAR

* Phobia - You can see in chat that the slime plush is making the person unconfortable
dreamseeker_bisB6ixtMt

</details>

## Changelog
:cl:
add: Added 5 new slime plushies
add: Added a rainbow flush plush
tweak: Removed dehydrated carps from the plushes crate
add: Added a new contraband crate: "Plushie Crate Without Moth Plushies". Dehydrated carps included, moth plushes excluded.
tweak: Alien phobia now includes slime plushes (it already included slime mobs).
spellcheck: There's no such thing as plushys, they're plushies now.
/:cl:

### CODE REVIEW

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -3108,9 +3108,26 @@
...
+	num_contained = 5
+	contains = list()
+	crate_type = /obj/structure/closet/crate/wooden
+	crate_name = "plushie crate"
+
+/datum/supply_pack/costumes_toys/randomised/plush_no_moths/fill(obj/structure/closet/crate/C)
+	var/plush_nomoth
+	for(var/i in 1 to num_contained)
+		plush_nomoth = pick(subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth))
+		new plush_nomoth(C)
```
Comment (EvilDragonfiend) [inline code/modules/cargo/packs.dm]: cache this subtype list into a datum variable. You're recalculating a list every this loop.
or at least you should declare a temporary variable right above of line 3127

File: code/modules/cargo/packs.dm [hunk 2]
```diff
@@ -3108,9 +3108,26 @@
 /datum/supply_pack/costumes_toys/randomised/plush/fill(obj/structure/closet/crate/C)
 	var/plush
 	for(var/i in 1 to num_contained)
-		plush = pick(subtypesof(/obj/item/toy/plush))
+		plush = pick(subtypesof(/obj/item/toy/plush) - /obj/item/toy/plush/carpplushie/dehy_carp)
```
Comment (EvilDragonfiend) [inline code/modules/cargo/packs.dm]: this too

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]

File: code/modules/cargo/packs.dm (hunk 1 above)
Comment (Gilgaxx) [reply code/modules/cargo/packs.dm]: I'm not sure what "cache this subtype list into a datum variable" means, I'm still a bit new to this, could you give an example or explain with layman terms?
Also, line 3126 is `var/plush_nomoth`. Is that not a declaration of a temporary variable?

Comment (EvilDragonfiend) [inline code/modules/cargo/packs.dm]: These lines actually mean:
```dm
	var/plush_nomoth
	for(var/i in 1 to num_contained)
		var/_temporary_list = subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth)
		plush_nomoth = pick(_temporary_list )
		new plush_nomoth(C)
```
You can see `var/_temporary_list = subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth)` is in the loop. This means you're creating a list of items every time this loop is continued.
but calculation can be done for the first time.
```dm
	var/plush_nomoth
	var/_temporary_list = subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth)
	for(var/i in 1 to num_contained)
		plush_nomoth = pick(_temporary_list )
		new plush_nomoth(C)
```
Now you don't calculate the list again and again.

Comment (EvilDragonfiend) [reply code/modules/cargo/packs.dm]: I made another comment to explain this.

Comment (Gilgaxx) [reply code/modules/cargo/packs.dm]: Oh, that makes sense! I'll change both of the crates so they work like that

Comment (Gilgaxx) [reply code/modules/cargo/packs.dm]: Alright a new commit should be up.

<details>
<summary>And here is me spawning one of each crate with the cargo spawn verb:</summary>

dreamseeker_N2psD7B3rX
</details>
The runtime happened a few seconds after loading and I don't believe it's related to this PR, do correct me if I'm wrong!

Comment (EvilDragonfiend) [reply code/modules/cargo/packs.dm]: yeah it's unrelated. no worries.

**Review by EvilDragonfiend — APPROVED** [review]
Pigeon requested me checking if we can merge suicide procs (since slime and moth ones are the same), but I do not see it works here.
I will just approve it and let other maints look on it in this case.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**